code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -57,3 +59,15 @@
 				step_away(src, user, 15)
 				sleep(3)
 				step_away(src, user, 15)
+
+/mob/living/silicon/robot/hitby(atom/movable/AM, skipcatch = 0, hitpush = 1, blocked = 0)
+	var/obj/item/I
+	if(istype(AM, /obj/item))
+		I = AM
+	if(I)
```

Comment:
Compose these conditionals together. Additionally, make the `istype` negated and do a parent call inside, as above.